### ISSUE TITLE
as.owin problem

### ISSUE THREAD

**Issue by hcommenges** [spatstat/spatstat#79]
Dear spatstat developpers,

I can't understand how the function as.owin() transforms an sp SpatialPolygon object. 
1. If I read a shapefile with maptools as.owin() works well. 
2. If I read a shapefile with RGDAL as.owin() doesn't work (Error:Can't interpret W as a window)
3. If I read a shapefile with maptools, then store it as Rds, then load it back with readRDS(), the function as.owin() doesn't work (Error:Can't interpret W as a window)

If I check the differences between 1 and 2, the objects are identical. If I check the differences between 1 and 3, the objects are identical. And also, the original shapefile is very simple and the geometry is perfectly consistent. I've seen a StackOverflow issue (https://stackoverflow.com/questions/19183105/error-in-as-owin-handling-shape-files-in-the-spatstat-package-r) but no pertinent answer in it. I'm pretty sure it is not a topology problem. Could you help me with that?

Thanks

Hadrien

**Comment by rubak** [comment]
It will make it a lot easier to troubleshoot this if you can provide a reproducible example. Can you please do that?

**Comment by baddstats** [comment]
You will need the package `maptools` to be loaded, in order to do the conversion. Please check that `maptools` was loaded in each of the 3 cases you mention.

`as.owin` is generic. There are methods for `as.owin` inside the `spatstat` package, and other methods inside the `maptools` package, and maybe elsewhere. The `spatstat` package does not support classes from other packages like `sp` or 'sf`. Conversion from `sp` classes to `spatstat` classes is provided by `maptools`.

**Comment by hcommenges** [comment]
Resolved, thanks! 
It was a conversion from a `sp` object and I thought that `as.owin()` was a specific `spatstat` function that could deal with `sp` class. And I found an explicit warning in the vignette about shapefiles: "Both packages maptools and spatstat must be loaded in order to convert the data". Of course it's my mistake but I find that the links between both packages are difficult to understand. Especially for users like me: frequent users of `spatstat` functions who never use any `maptools` function.
